src/test/regress/sql/subselect_gp.sql

```diff
@@ -1248,3 +1248,124 @@ select count(*) from t1;
 
 drop table t2;
 drop table t1;
+
+-- Test correlated SubPlans containing writable operation are
+-- planned and executed correctly. The result of modifying operations
+-- should be broadcasted (or focused) and materialized.
+-- start_ignore
+drop table if exists t1;
+drop table if exists t2;
+--end_ignore
+create table t1(i int) distributed by (i);
+create table t2(i int) distributed by (i);
+insert into t2 values (1), (2);
+
+explain (costs off)
+with cte as
+(insert into t1
+ select i from generate_series(1, 5) i
+ returning *)
+ select * from t2
```

Comment:
Remove please the space before "select" here and below. The select should be on the same level as with. It is not a part of with

Reply:
done